src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs

```diff
@@ -90,9 +90,12 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
                 config.UseMiddleware<HostWarmupMiddleware>();
             });
 
-            // This middleware must be registered before any other middleware depending on
-            // JobHost/ScriptHost scoped services.
-            builder.UseMiddleware<ScriptHostRequestServiceProviderMiddleware>();
+            builder.UseWhen(context => !context.Request.IsAdminResumeRequest(), config =>
+            {
+                // This middleware must be registered before any other middleware depending on
+                // JobHost/ScriptHost scoped services.
+                config.UseMiddleware<ScriptHostRequestServiceProviderMiddleware>();
```

Comment:
I find it interesting we would want this behavior on _any_ admin requests. But guess that is how it is, big risk to change that further.

Reply:
Yeah I was thinking about that too -- but this seems a lot safer right now. We may end up there.